libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -202,11 +202,16 @@ class CodeTransform: public boost::static_visitor<>
 	void operator()(assembly::VariableDeclaration const& _varDecl)
 	{
 		int height = m_state.assembly.deposit();
+		int expectedItems = _varDecl.variables.size();
 		boost::apply_visitor(*this, *_varDecl.value);
-		expectDeposit(1, height);
-		auto& var = boost::get<Scope::Variable>(m_scope.identifiers.at(_varDecl.variable.name));
-		var.stackHeight = height;
-		var.active = true;
+		expectDeposit(expectedItems, height);
+		int index = expectedItems;
+		for (auto const& variable: _varDecl.variables)
+		{
+			auto& var = boost::get<Scope::Variable>(m_scope.identifiers.at(variable.name));
+			var.stackHeight = --index;
```

Comment:
I think this changes semantics. `height` can be much more than just the current expression stack.
The following could work:
```
index--;
var.stackHeight = height - index;
```